src/storage/tile.cpp

```diff
@@ -67,6 +67,14 @@ Tile::Tile(BackendType backend_type, TileGroupHeader *tile_header,
   // if (schema.IsInlined() == false) {
   pool = new type::EphemeralPool();
   //}
+
+  // Record memory deallocation
+  if (table_id != INVALID_OID &&
+      static_cast<StatsType>(settings::SettingsManager::GetInt(
```

Comment:
I see that this condition is used in many places in your code. Can this logic be moved to a function somewhere?